.github/workflows/dotnet-code-metrics.yml

```diff
@@ -0,0 +1,36 @@
+name: '.NET code metrics'
+
+on:
+  push:
```

Comment:
Do you want to add running this on PR open & commit? (Open question, I can argue both sides. I'd like to hear your thoughts.

Reply:
The GitHub Action that it will be calling into, evaluates code metrics for a targeted `dir`. It will not do code metrics on all the samples, that would be a LOT. The action creates a PR to maintain a CODE_METRICS.md file. I like `on push` here, since it is final. I don't think we'd want a PR on ever rev of a commit.